istioctl/cmd/proxyconfig.go

```diff
@@ -457,8 +467,11 @@ func allConfigCmd() *cobra.Command {
 					if err != nil {
 						return err
 					}
-
-					if isZtunnelPod(podName, podNamespace) {
+					ztunnelPod, err := isZtunnelPod(podName, podNamespace)
```

Comment:
In `isZtunnelPod`, we initiate a new client, and in `extractConfigDump` and `extractZtunnelConfigDump`, we also initiate a new client to obtain the config dump. I think we can combine these two together.

Reply:
Maybe rename `isZtunnelPodKubeClient ` to `proxyConfigKubeClient` and use it everywhere in the proxyconfig command.

Reply:
At first glance this seems a bit tricky because I'm not sure it's guaranteed if `proxyConfigKubeClient` is set for each of these instances where we create a kube client.

Right now it's only set when `isZtunnelPod` is called. And not every command calls `isZtunnelPod` first :thinking:

Going to have to find a way to populate that value first thing.. Maybe a few functions will still need to initialize but maybe they would just set the `proxyConfigKubeClient` value then.

I'll try some things and push up the change shortly.

Reply:
here's just one example of what I'm referring to:

```go
func workloadConfigCmd() {
.
.
.
    RunE: func(c *cobra.Command, args []string) error {
			var configWriter *ztunnelDump.ConfigWriter
			var err error
			if len(args) == 1 {
				if podName, podNamespace, err = getComponentPodName(args[0]); err != nil {
					return err
				}
				ztunnelPod, kubeClientErr := isZtunnelPod(podName, podNamespace)
				if kubeClientErr != nil {
					return fmt.Errorf("failed to determine if pod is a zTunnel pod: %v", kubeClientErr)
				}
				if !ztunnelPod {
					return fmt.Errorf("workloads command is only supported by ztunnel proxies: %v", podName)
				}
				configWriter, err = setupZtunnelConfigDumpWriter(podName, podNamespace, c.OutOrStdout())
			} else {
				configWriter, err = setupFileZtunnelConfigdumpWriter(configDumpFile, c.OutOrStdout())
			}
```

`getComponentPodName` calls `getPodNameWithNamespace` which currently initializes it's own kubeClient. Replacing that with `proxyConfigKubeClient` wouldn't work if `isZtunnelPod` wasn't called previously. Same issue if `len(args) != 1` and it goes down the path of calling `setupFileZtunnelConfigdumpWriter`.

Reply:
If the goal is now to de-duplicate client inits in the entire package, you probably want a standalone method (getKubeClient) and a package-level sync.Once. Each of the functions you listed should call that instead of initializing its own kubeCLient

Reply:
@keithmattix 

Yeah I was thinking something along those lines:

```go
var (
	loggerLevelString     = ""
	reset                 = false
	once                  sync.Once
	proxyConfigKubeClient kube.CLIClient
)

var getProxyConfigKubeClient = func(kubeconfig, configContext string) error {
	var err error
	once.Do(func() {
		proxyConfigKubeClient, err = kubeClient(kubeconfig, configContext)
	})
	return err
}

var isZtunnelPod = func(podName, podNamespace string) (bool, error) {
	err := getProxyConfigKubeClient(kubeconfig, configContext)
	if err != nil {
		return false, err
	}
	return ambientutil.IsZtunnelPod(proxyConfigKubeClient, podName, podNamespace), nil
}
```

And callers would do similar to what isZtunnelPod is doing:
```go
err := getProxyConfigKubeClient(kubeconfig, configContext)
if err != nil {
	return "", fmt.Errorf("failed to create Kubernetes client: %v", err)
}
```
Then use `proxyConfigKubeClient` as needed.

I'm currently testing it now and it's failing unit tests. So I'm figuring that out atm.

Reply:
Ok, `package-level sync.Once` is implying that we need to do this across the entire package. I was interpreting their comment as just needing to do it in this file for this PR.

Does it make sense to couple the two goals `package level de-duplication` _and_ `log level validity improvement`? Seems like the scope is growing past the initial goal of the PR.

Reply:
Good point; I'm in favor of a follow-up

Reply:
If the scope is expanding to the package level, I would prefer to do a follow-up as well.